.github/CHANGELOG.md

```diff
@@ -6,10 +6,16 @@
 
 ### Improvements
 
+* Update C++ and Python GitHub actions names to include the matrix info.
+  [(#717)](https://github.com/PennyLaneAI/pennylane-lightning/pull/717)
+
```

Comment:
I like this improvement, we should have more systematic naming in the CI. But why was this necessary in this PR?

Reply:
This is required to define proper CI requirements for PRs.